src/librustc/mir/mod.rs

```diff
@@ -1900,7 +1902,14 @@ pub fn print_miri_value<W: Write>(value: Value, ty: Ty, f: &mut W) -> fmt::Resul
         (Value::ByVal(PrimVal::Bytes(bits)), &TyFloat(ast::FloatTy::F64)) =>
             write!(f, "{}f64", Double::from_bits(bits)),
         (Value::ByVal(PrimVal::Bytes(n)), &TyUint(ui)) => write!(f, "{:?}{}", n, ui),
-        (Value::ByVal(PrimVal::Bytes(n)), &TyInt(i)) => write!(f, "{:?}{}", n as i128, i),
+        (Value::ByVal(PrimVal::Bytes(n)), &TyInt(i)) => {
+            let bits = ty::tls::with(|tcx| {
+                Integer::from_attr(tcx, SignedInt(i)).size().bits()
```

Comment:
cc @oli-obk this might be a useful way to handle some of the usecases that you needed `layout_of` for.

Reply:
which was exactly what was happening in the body of `scalar_size` ;)

There's two cases where we can use this trick, but all other cases would need explicit matching and such, which I wrapped in `scalar_size`.